packages/flutter/lib/src/material/input.dart

```diff
@@ -54,18 +51,21 @@ class Input extends StatefulWidget {
     this.isDense: false,
     this.autofocus: false,
     this.maxLines: 1,
-    this.formField,
     this.onChanged,
     this.onSubmitted,
   }) : super(key: key);
 
-  /// The text of the input field.
-  ///
-  /// If the [Input] is in a [Form], this is the initial value only.
-  ///
-  /// Otherwise, this is the current value, and must be updated every
-  /// time [onChanged] is called.
-  final InputValue value;
+  /// The editable text displayed in the input field. If this is null, the
+  /// [Input] manages it for you.
+  final String text;
+
+  /// The range of editable text that is currently selected. If this is null,
+  /// the [Input] manages it for you.
+  final TextSelection selection;
+
+  /// The range of editable text that is still being composed. If this is null,
+  /// the [Input] manages it for you.
+  final TextRange composing;
```

Comment:
Shouldn't these two be replaced by an InputValue? Maybe I misunderstood your "half-way" proposal. So long as you have to call out each part of the InputValue separately, the problem still exists, no?